packages/flame/lib/src/game/flame_game.dart

```diff
@@ -86,6 +86,24 @@ class FlameGame extends ComponentTreeRoot
     setMounted();
   }
 
+  /// Setting to true will remove all children and process any remaining life
+  /// cycle events.
+  bool disposeAll = false;
```

Comment:
This name should reflect when it is doing the disposal, since there is no natural way of doing this.

Reply:
Maybe `disposeAllOnRemove`?

Reply:
What I really wanted was a named optional parameter for onRemove but that would then be a breaking change, so I was trying to come up with a way to make it accessible and clear.  Because google says `dispose` is permanent and there is no coming back, I wanted to keep it like that because if set to true, then everything gets wiped.  It could be `disposeAllOnRemove` but you still have to call super to invoke it.  

I didn't make it `@mustcallsuper` because after talking with @spydon, that would be breaking and again, this is really optional and in their opinion, cleaning up your game state is really the users decision and not something Flame should be deciding.  This gives them the option, but does not force it on the user.